src/route.js

```diff
@@ -14,41 +14,71 @@ import PlaybackPage from 'containers/PlayBackPage';
 import EditQuestionPage from 'containers/EditQuestionPage';
 import NotFoundPage from 'containers/NotFoundPage';
 import CandidateListPage from 'containers/CandidateListPage';
+import { checkPermission } from 'utils/auth';
 
 const { PUBLIC_URL } = process.env;
 
-export default () => (
-  <Router basename={PUBLIC_URL}>
-    <Switch>
-      <Route path="/admin">
-        <Admin>
-          <Route exact path="/admin" component={MainPage} />
-          <Route
-            exact
-            path="/admin/dispatch/:roomId"
-            component={DispatchPage}
-          />
-          <Route
-            exact
-            path="/admin/playback/:testId"
-            component={PlaybackPage}
-          />
-          <Route
-            exact
-            path="/admin/add"
-            render={props => <EditQuestionPage {...props} type="add" />}
-          />
-          <Route
-            exact
-            path="/admin/edit"
-            render={props => <EditQuestionPage {...props} type="edit" />}
-          />
-          <Route exact path="/admin/candidates" component={CandidateListPage} />
-        </Admin>
-      </Route>
-      <Route exact path="/exam/:roomId" component={ExamPage} />
-      <Redirect exact from="/" to="/admin" />
-      <Route component={NotFoundPage} />
-    </Switch>
-  </Router>
-);
+class route extends Component {
+  state = {
+    shouldLogin: true,
+    hasPermission: false,
+  };
+
+  async componentDidMount () {
+    await this.checkPermission()
+  }
+
+  checkPermission = async () => {
+    let isAdmin = await checkPermission()
+    this.setState({ hasPermission: isAdmin, shouldLogin: !localStorage.username });
+  };
```

Comment:
Hey @kennychang0603, why do we use both **shouldLogin** and **hasPermission** to  determine if user can go to **Admin** page instead of using **hasPermission** only? thanks.